File: 2021 marzec/c++/2.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <algorithm>
using namespace std;

string kraj[50], miasto[50];
int sklep[50][140];
pair <int, string> ans[50];

int main()
{
    ifstream in("galerie.txt");
    ofstream outa("wynik4_2a.txt");
    ofstream outb("wynik4_2b.txt");
    
    for (int i = 0; i < 50; i++)
    {
        in >> kraj[i] >> miasto[i];
        for (int j = 0; j < 140; j++)
            in >> sklep[i][j];
    }

    for (int i = 0; i < 50; i++)
    {
        int p = 0, l = 0;
        for (int j = 0; j < 140; j += 2)
        {
            int cur = sklep[i][j] * sklep[i][j + 1];
            if (cur == 0)
                break;
            p += cur;
            l++;
        }
        ans[i] = {p, miasto[i]};
        outa << miasto[i] << ' ' << p << ' ' << l << '\n';
    }

    sort(ans, ans + 50);
    outb << ans[49].second << ' ' << ans[49].first << '\n';
    outb << ans[0].second << ' ' << ans[0].first;
}

File: 2021 marzec/python/2.py
with open("galerie.txt", "r") as f:
    galerie = f.read().strip().split('\n')

kraj, miasto, lokal = [], [], []
for g in galerie:
    g = g.split()
    kraj.append(g[0])
    miasto.append(g[1])
    lokal.append(list(map(int, g[2:])))

ans = []
with open("wynik4_2a.txt", "w") as f:
    for i in range(50):
        p, l = 0, 0
        for j in range(0, 140, 2):
            cur = lokal[i][j] * lokal[i][j + 1]
            if cur == 0:
                break
            p += cur
            l += 1

        f.write(miasto[i] + ' ' + str(p) + ' ' + str(l) + '\n')
        ans.append((p, miasto[i]))

ans.sort()
with open("wynik4_2b.txt", "w") as f:
    f.write(ans[49][1] + ' ' + str(ans[49][0]) + '\n')
    f.write(ans[0][1] + ' ' + str(ans[0][0]))